module2/Homework/philosopher.h

```diff
@@ -0,0 +1,100 @@
+#ifndef _PHILOSOPHER_H_
+#define _PHILOSOPHER_H_
+
+#include <iostream>
+#include <mutex>
+#include <shared_mutex>
+#include <condition_variable>
+#include <algorithm>
+#include <random>
+#include <memory>
+#include <thread>
+#include <chrono>
+#include <sstream>
+#include <functional>
+
+
+class EatTimeEngine
+{
+private:
+    std::mt19937 m_generator;
+    std::uniform_int_distribution<int> m_distribution;
+    static EatTimeEngine *pinstance_;
```

Comment:
polecam implementować singletony Meyersa - https://stackoverflow.com/questions/17712001/how-is-meyers-implementation-of-a-singleton-actually-a-singleton

Najprostsze i bezpieczne bez dodatkowych mechanizmów.